Meilidown/Controllers/DocumentsController.cs

```diff
@@ -48,9 +54,61 @@ public DocumentsController(IIndexingService indexingService)
             }
         }
 
+
+        var orderFiles = _sourcesProvider.GetSources()
+                                         .SelectMany(source => source.FindFiles(new("\\.order")))
+                                         .ToDictionary(sf => sf.RelativePath.Split('/')[0..^1],
+                                                       sf => sf);
+
+        await SetOrder(tree, orderFiles);
+
         return tree;
     }
 
+    private async Task SetOrder(Dictionary<string, NavNode> tree, Dictionary<string[], SourceFile> orderFiles)
+    {
+        int i = 0;
+
+        foreach (var treeNode in tree)
+            await SetOrderByParent(treeNode.Value, orderFiles, i++);
+    }
+
+    [NonAction]
+    private async Task SetOrderByParent(NavNode node, Dictionary<string[], SourceFile> orderFiles, int index)
+    {
+        node.order = index;
+
+        if (node.children != null)
+        {
+            int childIndex = 0;
+
+            // recurse all children
+            foreach (var treeNode in node.children)
+                await SetOrderByParent(treeNode.Value, orderFiles, childIndex++);
+
+            var path = node.location.Split('/');
+
+            // if this particular folder has a .order, override the order
+            var orderFile = orderFiles.SingleOrDefault(of => of.Key.SequenceEqual(path));
+
+            if (orderFile.Value != null)
+            {
+                var lines = await System.IO.File.ReadAllLinesAsync(orderFile.Value.AbsolutePath);
+
+                foreach (var item in node.children)
+                {
+                    int newIndex = Array.IndexOf(lines, item.Key);
+                    item.Value.order = newIndex;
+
+                    if (newIndex < 0)
+                        _logger.LogDebug($"Hiding {item.Value.location}, according to `.order`");

```

Comment:
Maybe the whole order logic should be moved to the indexing service, so certain files can be hidden entirely, not just the nav tree. This would also make the dependecy on `ISourcesProvider` unnecessary. (Ideally, the controller should contain as little logic as possible anyways)

Reply:
I think we should have a separate service and remove the `order` from indexing altogether. The order depends on the siblings with the current tree node, whereas the index is currently linear. We _could_ put the `.order` files in the index as well, and then ask it for all files that match a path, but that seems silly to me.

Reply:
Yeah, makes sense. The order is only important for the nav tree